kimchi/src/verifier.rs

```diff
@@ -480,6 +480,42 @@ where
     }
 }
 
+fn check_proof_evals_len<G>(proof: &ProverProof<G>)
+where
+    G: KimchiCurve,
+    G::BaseField: PrimeField,
+{
+    let ProofEvaluations {
+        w,
+        z,
+        s,
+        coefficients,
+        lookup,
+        generic_selector,
+        poseidon_selector,
+    } = &proof.evals;
+
+    let check_eval_len = |eval: &PointEvaluations<Vec<_>>| {
+        assert!(eval.zeta.len() == 1);
+        assert!(eval.zeta_omega.len() == 1);
```

Comment:
these should return an error instead of panicking here : o

Reply:
Same as above